lib/Runtime/PlatformAgnostic/UnicodeText.h

```diff
@@ -0,0 +1,182 @@
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#pragma once
+
+#include "CommonPal.h"
+#include "Core/CommonTypedefs.h"
+
+namespace PlatformAgnostic
+{
+    namespace UnicodeText
+    {
+        // This structure is used by a subset of APIs where 
+        // errors are expected. In that case, ApiError is the last out
+        // parameter.
+        enum ApiError
+        {
+            NoError,
+            InvalidParameter,
+            InvalidUnicodeText,
+            InsufficientBuffer,
+            UntranslatedError
+        };
+
+        // The form to use for NormalizeString
+        // Intentionally compatible with the NORM_FORM Win32 enum
+        enum NormalizationForm
+        {
+            NormalizationOther = 0,    // Not supported
+            NormalizationC = 0x1,  // Each base plus combining characters to the canonical precomposed equivalent.
+            NormalizationD = 0x2,  // Each precomposed character to its canonical decomposed equivalent.
+            NormalizationKC = 0x5,  // Each base plus combining characters to the canonical precomposed
+                                       //   equivalents and all compatibility characters to their equivalents.
+            NormalizationKD = 0x6   // Each precomposed character to its canonical decomposed equivalent
+                                       //   and all compatibility characters to their equivalents.
+        };
+
+        // Mapping of a unicode codepoint to a class of characters
+        // Used by the legacy API
+        enum CharacterClassificationType: byte
+        {
+            ClassInvalid,
+            ClassLetter,
+            ClassWhitespace,
+            ClassNewLine,
+            ClassDigitOrPunct
+        };
+
+        // Used by the legacy API for mapping
+        // a codepoint to a set of flags
+        enum CharacterTypeFlags : uint32
+        {
+            UnknownChar = 0x0,
+            IdChar = 0x01,
+            IdLeadChar = 0x02,
+            HexChar = 0x04,
+            DecimalChar = 0x08,
+            SpaceChar = 0x10,
+            LineFeedChar = 0x20,
+
+            LineCharGroup = SpaceChar | LineFeedChar,
+            LetterCharGroup = IdChar | IdLeadChar,
+            HexCharGroup = IdChar | IdLeadChar | HexChar,
+            DecimalCharGroup = IdChar | DecimalChar,
+        };
+
+        // Parameters for APIs to change a strings case
+        enum CaseFlags
+        {
+            CaseFlagsUpper,
+            CaseFlagsLower
+        };
+
+        // Subset of the unicode general category listed in Table 4-9 of the Unicode 8.0 Spec
+        // We map the unicode categories into "category classes" at the granularity that the 
+        // user of this data needs to discriminate the category at.
+        enum UnicodeGeneralCategoryClass
```

Comment:
Welp, you're right- I'll change to using enum class. I keep losing track of which portions of C++11 2013 supports